crates/virtio-queue/src/queue.rs

```diff
@@ -741,4 +741,35 @@ mod tests {
             }
         }
     }
+
+    #[test]
+    fn test_regression_ordering_avail_used() {
+        // This is a regression test for a fuzzing finding that was triggering a panic when
+        // `availd_idx` and `used_idx` were called with an unsupported ordering.
+        let m = &GuestMemoryMmap::<()>::from_ranges(&[(GuestAddress(0), 0x10000)]).unwrap();
+        let vq = MockSplitQueue::new(m, 16);
+
+        let q = vq.create_queue(m);
+        let supported_ordering = [Ordering::SeqCst, Ordering::Acquire, Ordering::Relaxed];
+        for ord in supported_ordering {
+            assert!(q.avail_idx(ord).is_ok());
+            assert!(q.used_idx(ord).is_ok());
+        }
+
+        // Let's store something dummy in `avail` & `used` so that `avail_idx` and `used_idx` can
```

Comment:
Shouldn't we do this before the calls of the two methods above? Also the comment says we store data in `avail` and `used`, but looks like we are only doing that for `avail`.

Reply:
Scratch that. I deleted the stores because actually we don't need them. The `Ok(0)` was returned in my local branch because of a bug that I had in the test code. I removed the store calls.